src/helpers/programHelper.js

```diff
@@ -451,6 +456,176 @@ getContributionWithProgramId(program_id, req) {
         });
       });
   }
+
+  copyCollections(data, channel, reqHeaders, cb) {
+    const rspObj = {};
+    const errObj = {
+      'loggerMsg': null,
+      'errCode': null,
+      'errMsg': null,
+      'responseCode': null
+    };
+  
+    if (!data.program_id || !data.config.collections || !data.content_types || !channel) {
+      errObj.errCode = programMessages.COPY_COLLECTION.COPY.MISSING_CODE;
+      errObj.errMsg = programMessages.COPY_COLLECTION.COPY.MISSING_MESSAGE;
+      errObj.responseCode = responseCode.CLIENT_ERROR;
+      errObj.loggerMsg = 'Error due to missing request or program_id or request collections or request allowed_content_types or channel'
+      cb(errObj, null);
+      return false;
+    }
+  
+    const collections = _.get(data, 'config.collections');
+    const collectionIds = _.map(collections, 'id');
+    const additionalMetaData = {
+      programId: _.get(data, 'program_id'),
+      allowedContentTypes: _.get(data, 'content_types'),
+      channel: channel,
+      openForContribution: false
+    };
+    
+    hierarchyService.filterExistingTextbooks(collectionIds, reqHeaders)
+      .subscribe(
+        (resData) => {
+          const consolidatedResult = _.map(resData, r => {
+            return {
+              result: r.data.result,
+              config: r.config.data
+            }
+          })
+  
+          const existingTextbooks = hierarchyService.getExistingCollection(consolidatedResult);
+          const nonExistingTextbooks = hierarchyService.getNonExistingCollection(consolidatedResult)
+
+          if (existingTextbooks && existingTextbooks.length > 0) {
+            hierarchyService.getHierarchy(existingTextbooks, reqHeaders)
+              .subscribe(
+                (originHierarchyResult) => {
+                  const originHierarchyResultData = _.map(originHierarchyResult, r => {
+                    return _.get(r, 'data')
+                  })
+                  const getCollectiveRequest = _.map(originHierarchyResultData, c => {
+                    let children = [];
+                    const cindex = collections.findIndex(r => r.id === c.hierarchy.content.identifier);
```

Comment:
Have you tested it once?  are we sure it is `c.hierarchy.content` .I am just wondering if the it is under the `hierarchy` property.

Reply:
@venkateshwarans Yes it is there